homeassistant/components/fitbit/__init__.py

```diff
@@ -34,7 +36,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except FitbitApiException as err:
         raise ConfigEntryNotReady from err
 
-    hass.data[DOMAIN][entry.entry_id] = fitbit_api
+    fitbit_config = config_from_entry_data(entry.data)
+    coordinator: FitbitDeviceCoordinator | None = None
+    if fitbit_config.is_allowed_resource(FitbitScope.DEVICE, "devices/battery"):
+        coordinator = FitbitDeviceCoordinator(hass, fitbit_api)
+        try:
+            await coordinator.async_config_entry_first_refresh()
+        except FitbitApiException as err:
```

Comment:
This exception is already caught in the update method of the coordinator, so should ne be possible to get here.

Reply:
Yeah, and the coordinator already handles raising `ConfigEntryNotReady` during first refresh.

Reply:
Fixed. I updated tests to exercise this case.